osf/migrations/0081_auto_20180215_1348.py

```diff
@@ -0,0 +1,90 @@
+                ('name', models.CharField(max_length=128)),
+                ('advisory_board', models.TextField(blank=True, default=b'')),
+                ('description', models.TextField(blank=True, default=b'')),
+                ('domain', models.URLField(blank=True, default=b'')),
+                ('domain_redirect_enabled', models.BooleanField(default=False)),
+                ('external_url', models.URLField(blank=True, null=True)),
+                ('email_contact', models.CharField(blank=True, max_length=200, null=True)),
+                ('email_support', models.CharField(blank=True, max_length=200, null=True)),
+                ('social_twitter', models.CharField(blank=True, max_length=200, null=True)),
+                ('social_facebook', models.CharField(blank=True, max_length=200, null=True)),
+                ('social_instagram', models.CharField(blank=True, max_length=200, null=True)),
+                ('footer_links', models.TextField(blank=True, default=b'')),
+                ('facebook_app_id', models.BigIntegerField(blank=True, null=True)),
+                ('example', models.CharField(blank=True, max_length=20, null=True)),
+                ('allow_submissions', models.BooleanField(default=True)),
+                ('share_publish_type', models.CharField(choices=[(b'Preprint', b'Preprint'), (b'Thesis', b'Thesis')], default=b'Preprint', help_text=b'This SHARE type will be used when pushing publications to SHARE', max_length=32, null=True)),
+                ('share_source', models.CharField(blank=True, max_length=200, null=True)),
+                ('share_title', models.TextField(blank=True, default=b'', null=True)),
+                ('additional_providers', django.contrib.postgres.fields.ArrayField(base_field=models.CharField(max_length=200), blank=True, default=list, null=True, size=None)),
+                ('access_token', osf.utils.fields.EncryptedTextField(blank=True, null=True)),
+                ('preprint_word', models.CharField(choices=[(b'preprint', b'Preprint'), (b'paper', b'Paper'), (b'thesis', b'Thesis'), (b'none', b'None')], default=b'preprint', max_length=10, null=True)),
+                ('subjects_acceptable', osf.utils.datetime_aware_jsonfield.DateTimeAwareJSONField(blank=True, default=list, encoder=osf.utils.datetime_aware_jsonfield.DateTimeAwareJSONEncoder, null=True)),
+                ('default_license', models.ForeignKey(blank=True, null=True, on_delete=django.db.models.deletion.CASCADE, related_name='default_license', to='osf.NodeLicense')),
+                ('licenses_acceptable', models.ManyToManyField(blank=True, related_name='licenses_acceptable', to='osf.NodeLicense')),
+            ],
+            options={
+                'abstract': False,
+            },
+            bases=(dirtyfields.dirtyfields.DirtyFieldsMixin, models.Model),
+        ),
+        migrations.RemoveField(
```

Comment:
Right before this line, we need to populate the `osf_abstractprovider` table with data from `osf_preprintprovider`, as Django isn't smart enough to do this on its own. This can be done relatively easily with a bit of SQL:

```
migrations.RunSQL(
    [
        """
        INSERT INTO osf_abstractprovider (list, of, columns, to, insert, into)
            SELECT columns, in, matching, order, to, list, above, and, 'osf.preprintprovder' as type
            FROM osf_preprintprovider
        """
    ], [
        """
        INSERT INTO osf_preprintprovider (things)
            SELECT things  -- everything but type, I'm lazy
            FROM osf_abstractprovider
        """
    ]
),
```

This gives the re-keying of `Subject` below something to point to, and should (hopefully) ensure that existing FKs pointing to `PreprintProvider` (e.g. `PreprintService.provider`) still point to the right record.

I'd recommend testing this locally by:
1. Ensure the existence of a few preprints across different providers (both with and without a custom taxonomy) while on `develop` or `master`
2. Check out this branch and running `python manage.py migrate`
3. Make sure things behave as they did -- subjects and preprints point to the correct provider, can submit new preprints with the same workflow, etc.